Source/DafnyLanguageServer/Language/SemanticTokens/LspSemanticTokensGeneratingVisitor.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.Boogie;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.Dafny.LanguageServer.Language.SemanticTokens;
+
+/// <summary>
+/// Visitor responsible for populating a table of document tokens.
+/// </summary>
+public class LspSemanticTokensGeneratingVisitor : SyntaxTreeVisitor {
```

Comment:
Would it be possible to replace this visitor with an interface `IHasHighlight { bool Highlight(DafnySemanticTokensBuilder builder)` that the individual Node classes can implement ?

Reply:
> Would it be possible to replace this visitor with an interface IHasHighlight { bool Highlight(DafnySemanticTokensBuilder builder) that the individual Node classes can implement ?

That would require to make DafnyCore aware of the language server, so I wouldn't go that route.